native/src/main/java/io/ballerina/lib/aws/sqs/listener/Listener.java

```diff
@@ -0,0 +1,189 @@
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package io.ballerina.lib.aws.sqs.listener;
+
+import io.ballerina.lib.aws.sqs.CommonUtils;
+import io.ballerina.lib.aws.sqs.client.NativeClientAdaptor;
+import io.ballerina.runtime.api.Environment;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.QueueDoesNotExistException;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageResponse;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Native implementation of the Ballerina AWS SQS Listener.
+ * Manages the lifecycle of SQS message polling and service invocation.
+ */
+public final class Listener {
+
+    public static final String NATIVE_SQS_CLIENT = "nativeClient";
+    private static final String NATIVE_POLLING_CONFIG = "native.polling.config";
+    private static final String NATIVE_SERVICES = "native.services";
+    private static final String NATIVE_STOPPED = "native.stopped";
+
+    private Listener() {
+    }
+
+    /**
+     * Initializes the SQS listener with the given configuration.
+     * Sets up the SQS client and required native data.
+     */
+    public static Object init(Environment env,
+            BObject bListener,
+            BMap<BString, Object> pollingConfig,
+            BMap<BString, Object> connectionConfig) {
+        try {
+            // Initialize the SQS client
+            Object err = NativeClientAdaptor.init(bListener, connectionConfig);
+            if (err instanceof BError) {
+                return err;
+            }
+            // Store polling config, an empty service map and the stopped flag
+            bListener.addNativeData(NATIVE_POLLING_CONFIG, new PollingConfig(pollingConfig));
+            bListener.addNativeData(NATIVE_SERVICES, new ConcurrentHashMap<String, Service>());
+            bListener.addNativeData(NATIVE_STOPPED, new AtomicBoolean(true));
+        } catch (Exception e) {
+            return CommonUtils.createError("Failed to initialize SQS listener: " + e.getMessage(), e);
+        }
```

Comment:
The `BError` will be thrown, which you can catch in the `catch` block. 

```suggestion
        try {
            // Initialize the SQS client
            NativeClientAdaptor.init(bListener, connectionConfig);
            // Store polling config, an empty service map and the stopped flag
            bListener.addNativeData(NATIVE_POLLING_CONFIG, new PollingConfig(pollingConfig));
            bListener.addNativeData(NATIVE_SERVICES, new ConcurrentHashMap<String, Service>());
            bListener.addNativeData(NATIVE_STOPPED, new AtomicBoolean(true));
        } catch (BError e) {
            return e;
        }catch (Exception e) {
            return CommonUtils.createError("Failed to initialize SQS listener: " + e.getMessage(), e);
        }
```

Reply:
Updated